x-pack/legacy/plugins/siem/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -86,33 +90,74 @@ export const signalRulesAlertType = ({
         ruleStatusSavedObjects,
         name,
       });
-      // set searchAfter page size to be the lesser of default page size or maxSignals.
-      const searchAfterSize =
-        DEFAULT_SEARCH_AFTER_PAGE_SIZE <= params.maxSignals
-          ? DEFAULT_SEARCH_AFTER_PAGE_SIZE
-          : params.maxSignals;
+
+      const searchAfterSize = Math.min(params.maxSignals, DEFAULT_SEARCH_AFTER_PAGE_SIZE);
+
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      let bulkIndexResult: any;
+
       try {
-        const inputIndex = await getInputIndex(services, version, index);
-        const esFilter = await getFilter({
-          type,
-          filters,
-          language,
-          query,
-          savedId,
-          services,
-          index: inputIndex,
-        });
+        if (type === 'machine_learning') {
+          if (mlJobId == null || anomalyThreshold == null) {
+            throw new Error('Attempted to execute ML Rule, but missing jobId or anomalyThreshold');
+          }
```

Comment:
Similar to previous comment I don't know if you want to use `BadRequest` for this since it might not be related to a REST API call.

However, I would at least augment this with as much information as you can such as:
```ts
throw new Error(`Attempted to execute machine learning rule, but it is missing job id and/or anomaly threshold for rule id: "${ruleId}", name: "${name}", signals index: "${outputIndex}", job id: "${mlJobId}", anomaly threshold: "${anomalyThreshold}"`
);
```

Just in case later we encounter a bug or issue on a customer site and get logs we can hopefully figure out where the issue is coming from and fix them locally.